test/make_dist.py

```diff
@@ -61,6 +61,10 @@ def download_dist():
     This can happen because the current directory is not a git checkout, or it is
     a SHA which is not pushed/PRed, or there is no ~/.config/github-token available.
     '''
+    if os.path.exists("dist"):
```

Comment:
I think it would be cleaner to rework the logic in `make_dist()` below to default to calling `build_dist()` if `dist/` already exists, and only call `download()` if we're sure that's what we want to do.

Reply:
I was actually thinking about doing it the other way around: Right now most of the checks are inside download_dist() already, and only the NODE_ENV check is in the caller. Does that work for you as well?

Reply:
.. of course I meant: ... so I'd rather move the NODE_ENV check into download_dist() as well. It's the same kind of conditional check as "local changes" or "no tarball available" or "no token"

Reply:
Done.

Reply:
This thread is what i meant when I asked "where did the check go?"

At some point you moved it into `make_dist()` but now it's back here again...

Reply:
Right, I moved the one remaining check *from* make_dist() *to* download_dist().

Reply:
OK, moved to make_dist(). I can't say I like it, but I don't have a strong opinion about it :grin: Let's just fix this bug.